src/realm/query_engine.cpp

```diff
@@ -79,77 +79,119 @@ bool ParentNode::match(ConstObj& obj)
-        // For find_all(), the column below is a dummy and the caller sets it to nullptr. Hence, no data is being read
-        // from any column upon each query match (just a TableView.add(size_t index) is performed), and we pass
-        // nullable = false simply by convention. FIXME: Clean up all this.
-        if (nullable)
+    switch (TAction) {
+        case act_ReturnFirst: {
+            if (nullable)
+                m_column_action_specializer = &ThisType::column_action_specialization<act_ReturnFirst, ArrayIntNull>;
+            else
+                m_column_action_specializer = &ThisType::column_action_specialization<act_ReturnFirst, ArrayInteger>;
+            break;
+        }
+        case act_FindAll: {
+            // For find_all(), the column below is a dummy and the caller sets it to nullptr. Hence, no data is being
+            // read from any column upon each query match (just matchcount++ is performed), and we pass nullable =
+            // false simply by convention. FIXME: Clean up all this.
+            if (nullable)
+                REALM_ASSERT(false);
+            else
+                m_column_action_specializer = &ThisType::column_action_specialization<act_FindAll, ArrayInteger>;
+            break;
+        }
+        case act_Count: {
+            // For count(), the column below is a dummy and the caller sets it to nullptr. Hence, no data is being
+            // read from any column upon each query match (just matchcount++ is performed), and we pass nullable =
+            // false simply by convention. FIXME: Clean up all this.
+            if (nullable)
+                REALM_ASSERT(false);
+            else
+                m_column_action_specializer = &ThisType::column_action_specialization<act_Count, ArrayInteger>;
+            break;
+        }
+        case act_Sum: {
+            switch (col_id) {
+                case type_Int: {
+                    if (nullable)
+                        m_column_action_specializer = &ThisType::column_action_specialization<act_Sum, ArrayIntNull>;
+                    else
+                        m_column_action_specializer = &ThisType::column_action_specialization<act_Sum, ArrayInteger>;
+                    break;
+                }
+                case type_Float:
+                    m_column_action_specializer = &ThisType::column_action_specialization<act_Sum, ArrayFloat>;
+                    break;
+                case type_Double:
+                    m_column_action_specializer = &ThisType::column_action_specialization<act_Sum, ArrayDouble>;
+                    break;
+                case type_Decimal:
+                    m_column_action_specializer = &ThisType::column_action_specialization<act_Sum, ArrayDecimal128>;
+                    break;
+                default:
+                    REALM_ASSERT(false);
+                    break;
+            }
+            break;
+        }
+        case act_Min: {
+            switch (col_id) {
```

Comment:
don't we need timestamp support here for min/max?

Reply:
That could certainly be done - if requested.